http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/BackgroundProcess.scala

```diff
@@ -0,0 +1,84 @@
+package org.broadinstitute.dsde.workbench.leonardo.monitor
+
+import cats.Show
+import cats.effect.Async
+import cats.syntax.all._
+import fs2.Stream
+import org.broadinstitute.dsde.workbench.model.TraceId
+import org.broadinstitute.dsde.workbench.openTelemetry.OpenTelemetryMetrics
+import org.typelevel.log4cats.StructuredLogger
+
+import java.time.Instant
+import java.util.UUID
+import scala.concurrent.duration.FiniteDuration
+
+/**
+ * Defines a common interface that certain background processes should implement.
+ * This type of background processes do the following periodically:
+ * 1. Query the database for a list of candidates
+ * 2. Filter the candidates based on some criteria
+ * 3. Perform some action on the filtered candidates
+ */
+trait BackgroundProcess[F[_], A] {
+
+  /**
+   * The name of the background process. This is used mostly for log messages and metrics 
+   */
+  def name: String
+  def interval: FiniteDuration
+
+  /**
+   * Get a collection of candidates from the database
+   */
+  def getCandidates(now: Instant)(implicit
+    F: Async[F],
+    metrics: OpenTelemetryMetrics[F],
+    logger: StructuredLogger[F]
+  ): F[Seq[A]]
+
+  /**
+   * Perform some action on the candidate
+   */
+  def action(a: A, traceId: TraceId, now: Instant)(implicit F: Async[F]): F[Unit]
```

Comment:
This could be called `processCandidate` or `actOnCandidate` or something; it should be a verb.

Reply:
it's not a verb...this function describes the action that will happen

Reply:
function names are usually verbs; why break that pattern? `check()` is a verb, `getCandidates()` is a verb.

Reply:
I don't think function names need to always be verbs